x/auth/keeper/deterministic_test.go

```diff
@@ -118,3 +124,425 @@ func pubkeyGenerator(t *rapid.T) *rapid.Generator[secp256k1.PubKey] {
+			var account types.AccountI
+			err = suite.encCfg.InterfaceRegistry.UnpackAny(res.Accounts[i], &account)
+			suite.Require().NoError(err)
+
+			unpackedAccs[i] = account
+		}
+
+		sort.Slice(unpackedAccs, func(i2, j int) bool {
+			return unpackedAccs[i2].GetAccountNumber() < unpackedAccs[j].GetAccountNumber()
+		})
+
+		sort.Slice(prevRes, func(i2, j int) bool {
+			return prevRes[i2].GetAccountNumber() < prevRes[j].GetAccountNumber()
+		})
+
+		if prevRes != nil {
+			suite.Require().Equal(unpackedAccs, prevRes)
+		}
+
+		prevRes = unpackedAccs
+	}
+
+	for i := 0; i < len(prevRes); i++ {
+		suite.accountKeeper.RemoveAccount(suite.ctx, prevRes[i])
+	}
+}
+
+func (suite *DeterministicTestSuite) TestGRPCQueryAccounts() {
+
+	rapid.Check(suite.T(), func(t *rapid.T) {
+		numAccs := rapid.IntRange(1, 10).Draw(t, "accounts")
+		accs := make([]types.AccountI, numAccs)
+
+		for i := 0; i < numAccs; i++ {
+			pub := pubkeyGenerator(t).Draw(t, "pubkey")
+			addr := sdk.AccAddress(pub.Address())
+			accNum := rapid.Uint64Range(uint64(i*10+1), uint64(i*10+10)).Draw(t, "account-number") // to avoid collisions
+			seq := rapid.Uint64().Draw(t, "sequence")
+
+			acc1 := types.NewBaseAccount(addr, &pub, accNum, seq)
+			suite.accountKeeper.SetAccount(suite.ctx, acc1)
+			accs[i] = acc1
+		}
+
+		suite.runAccountsIterations(accs)
+	})
+
+	// Regression test
+	addr1 := sdk.MustAccAddressFromBech32("cosmos1892yr6fzlj7ud0kfkah2ctrav3a4p4n060ze8f")
+	pub1, err := hex.DecodeString("D1002E1B019000010BB7034500E71F011F1CA90D5B000E134BFB0F3603030D0303")
+	suite.Require().NoError(err)
+	accNum1 := uint64(107)
+	seq1 := uint64(0)
+
+	addr2 := sdk.MustAccAddressFromBech32("cosmos1j364pjm8jkxxmujj0vp2xjg0y7w8tyveuamfm6")
+	pub2, err := hex.DecodeString("01090C02812F010C25200ED40E004105160196E801F70005070EA21603FF06001E")
+	suite.Require().NoError(err)
+
+	accNum2 := uint64(100)
+	seq2 := uint64(10)
+
+	acc1 := types.NewBaseAccount(addr1, &secp256k1.PubKey{Key: pub1}, accNum1, seq1)
+	acc2 := types.NewBaseAccount(addr2, &secp256k1.PubKey{Key: pub2}, accNum2, seq2)
+
+	suite.accountKeeper.SetAccount(suite.ctx, acc1)
+	suite.accountKeeper.SetAccount(suite.ctx, acc2)
+
+	suite.runAccountsIterations([]types.AccountI{acc1, acc2})
+}
+
+func (suite *DeterministicTestSuite) runAccountAddressByIDIterations(id int64, prevRes string) {
+	for i := 0; i < 1000; i++ {
+		res, err := suite.queryClient.AccountAddressByID(suite.ctx, &types.QueryAccountAddressByIDRequest{Id: id})
+		suite.Require().NoError(err)
+		suite.Require().NotNil(res)
+
+		if prevRes != "" {
+			suite.Require().Equal(res.AccountAddress, prevRes)
+		}
+
+		prevRes = res.AccountAddress
+	}
+}
+
+func (suite *DeterministicTestSuite) TestGRPCQueryAccountAddressByID() {
+	rapid.Check(suite.T(), func(t *rapid.T) {
+		pub := pubkeyGenerator(t).Draw(t, "pubkey")
+		addr := sdk.AccAddress(pub.Address())
+
+		// TODO change this to draw uint64
+		accNum := rapid.Uint32().Draw(t, "account-number")
```

Comment:
Why uint32 here?

Reply:
```
res, err := suite.queryClient.AccountAddressByID(suite.ctx, &types.QueryAccountAddressByIDRequest{Id: id})
```
at this line `&types.QueryAccountAddressByIDRequest` accepting `int64` (which can accept -ve values) instead of `uint64`

if we draw `uint64` here and convert it to `int64` it'll go to -ve values because of range overflow.

I think we have to create an issue to change `&types.QueryAccountAddressByIDRequest` to accept only `uint64`, wdyt?

Reply:
Unfortunately I think we might need to revert https://github.com/cosmos/cosmos-sdk/pull/13411#issuecomment-1268715058.

I would propose to use `rapid.Int64()` directly here, instead of Uint32 then casting to Int64

Reply:
> Unfortunately I think we might need to revert https://github.com/cosmos/cosmos-sdk/pull/13411#issuecomment-1268715058.

Can you do that please